### PR TITLE
Model export

### PR DESCRIPTION
**Pull request by tejfel**

### CODE REVIEW

**Comment by ntrncic** [comment]
please check your tests. they're failing

File: EquiTrack/partners/exports.py [hunk 1]
```diff
@@ -316,3 +216,120 @@ def fill_row(self, pca, row):
...
+        return ', '.join(supply_names)
+
+
+class GovernmentExport(resources.ModelResource):
+    cash_transfer = resources.Field()
+    year = resources.Field()
+
+    class Meta:
+        model = GovernmentIntervention
+        fields = ('number', 'partner__name', 'result_structure__name', 'sectors', 'cash_transfer',
```
Comment (ntrncic) [inline EquiTrack/partners/exports.py]: the fields come out backwards in excel (cash transfer, sectors, result_structure etc..)

File: EquiTrack/partners/tests/test_exports.py [hunk 2]
```diff
@@ -1,10 +1,42 @@
...
+class TestModelExport(APITenantTestCase):
+    def setUp(self):
+        super(TestModelExport, self).setUp()
+        self.unicef_staff = UserFactory(is_staff=True)
+        self.partner = PartnerFactory()
+        self.agreement = AgreementFactory(partner=self.partner)
+        # This is here to test partner scoping
+        AgreementFactory()
+        # self.intervention = PartnershipFactory(partner=self.partner,
+        #                                        agreement=self.agreement)
```
Comment (ntrncic) [inline EquiTrack/partners/tests/test_exports.py]: can we add the Intervention factory and government_intervention factory and then run the tests see if a row is returned

File: EquiTrack/partners/tests/test_views.py [hunk 3]
```diff
@@ -49,8 +49,8 @@ def test_api_agreements_list(self):
         response = self.forced_auth_req('get', '/api/partners/'+str(self.intervention.partner.id)+'/agreements/', user=self.unicef_staff)
 
         self.assertEquals(response.status_code, status.HTTP_200_OK)
-        # the length of this list should be 1
-        self.assertEquals(len(response.data), 1)
+        # the length of this list should be 0
```
Comment (ntrncic) [inline EquiTrack/partners/tests/test_views.py]: should create an agreement for the partner and check if the length is 1

File: EquiTrack/partners/views.py [hunk 4]
```diff
@@ -282,6 +284,33 @@ def retrieve(self, request, partner_pk=None, pk=None):
             status=status.HTTP_200_OK
         )
 
+    @list_route(methods=['get'])
```
Comment (ntrncic) [inline EquiTrack/partners/views.py]: I don't think we need this. this isn't an excel export.

File: EquiTrack/partners/views.py [hunk 5]
```diff
@@ -806,6 +848,17 @@ def create(self, request, *args, **kwargs):
             headers=headers
         )
 
+    @list_route(methods=['get'])
```
Comment (ntrncic) [inline EquiTrack/partners/views.py]: don't need this exported into excel. please remove

Comment (tejfel) [inline EquiTrack/partners/views.py]: This should be the export for partner. Should we put this somewhere else?

File: EquiTrack/partners/tests/test_views.py (hunk 3 above)
Comment (tejfel) [inline EquiTrack/partners/tests/test_views.py]: done in the other pr (partner scoping)

File: EquiTrack/partners/exports.py (hunk 1 above)
Comment (tejfel) [inline EquiTrack/partners/exports.py]: check

File: EquiTrack/partners/views.py [hunk 6]
```diff
@@ -283,6 +286,36 @@ def retrieve(self, request, partner_pk=None, pk=None):
...
+    permission_classes = (PartnerPermission,)
+    filter_backends = (PartnerScopeFilter, GovernmentInterventionExportFilter,)
+
+    @list_route(methods=['get'])
+    def export(self, request, partner_pk):
+        queryset = self.get_queryset()
+        queryset = self.filter_queryset(queryset)
+        dataset = GovernmentExport().export(queryset)
+
+        response = HttpResponse('application/vnd.ms-excel')
```
Comment (ntrncic) [inline EquiTrack/partners/views.py]: please change to content_type='application/vnd.ms-excel'
I was getting a unicode error

File: EquiTrack/partners/views.py [hunk 7]
```diff
@@ -360,6 +393,17 @@ def retrieve(self, request, partner_pk=None, pk=None):
             status=status.HTTP_200_OK
         )
 
+    @list_route(methods=['get'])
+    def export(self, request, partner_pk):
+        queryset = self.get_queryset()
+        queryset = self.filter_queryset(queryset)
+        dataset = InterventionExport().export(queryset)
+
+        response = HttpResponse('application/vnd.ms-excel')
```
Comment (ntrncic) [inline EquiTrack/partners/views.py]: please change to content_type='application/vnd.ms-excel'
I was getting a unicode error

File: EquiTrack/partners/views.py [hunk 8]
```diff
@@ -293,8 +293,8 @@ def export(self, request, partner_pk):
         dataset = AgreementExport().export(queryset)
 
         response = HttpResponse(content_type='application/vnd.ms-excel')
-        response['Content-Disposition'] = 'attachment; filename="ModelExportAgreements.xlsx"'
-        response.write(dataset.xlsx)
+        response['Content-Disposition'] = 'attachment; filename="ModelExportAgreements.xls"'
+        response.write(dataset.xls)
```
Comment (ntrncic) [inline EquiTrack/partners/views.py]: why change the format to xls?

Comment (tejfel) [inline EquiTrack/partners/views.py]: It's changed back. I had some fight with the tablib library and used xlrd till I figured that out.

File: EquiTrack/partners/views.py (hunk 5 above)
Comment (robertavram) [inline EquiTrack/partners/views.py]: you can just switch it to a detail_route.. even better to get away from viewsets.. we can handle routing manually...

**Comment by robertavram** [comment]
@ntrncic i believe this is ready until moving to the new exports... please update form develop
and merge when ready

File: EquiTrack/partners/views.py (hunk 5 above)
Comment (ntrncic) [inline EquiTrack/partners/views.py]: @robertavram if it's a detail_route then requires PK due to views magic. so leaving it as list_route

**Comment by ntrncic** [comment]
must update from develop.
change related set usage for PartnerStaffMember and PCAGrant.

**Pull request merged**